File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/java-chatgpt-A/LoadApplicationSettingsYAMLDetailed-chatgpt.java
import org.yaml.snakeyaml.Yaml;

import java.io.FileInputStream;
import java.io.InputStream;
import java.util.LinkedHashMap;
import java.util.Map;

public class ConfigLoader {

    public static void main(String[] args) {
        try (InputStream input = new FileInputStream("config.yml")) {

            Yaml yaml = new Yaml();

            // Load YAML configuration
            Map<String, Object> loadedConfig = yaml.load(input);

            // Store configuration values in memory
            Map<String, Object> config = new LinkedHashMap<>();
            config.putAll(loadedConfig);

            // Display loaded settings
            System.out.println("=== Application Settings ===");

            for (Map.Entry<String, Object> entry : config.entrySet()) {
                System.out.println(entry.getKey() + " = " + entry.getValue());
            }

            System.out.println("----------------------------");
            System.out.println("Total Settings: " + config.size());

        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/python-chatgpt-A/load_application_settings_yaml_detailed-chatgpt.py
import yaml

def load_config(yaml_file):
    # Load YAML configuration
    with open(yaml_file, "r", encoding="utf-8") as file:
        loaded_config = yaml.safe_load(file)

    # Store configuration values in memory
    config = dict(loaded_config)

    return config

def display_config(config):
    print("=== Application Settings ===")

    for key, value in config.items():
        print(f"{key} = {value}")

    print("-" * 30)
    print(f"Total Settings: {len(config)}")

if __name__ == "__main__":
    config = load_config("config.yml")
    display_config(config)